### PR TITLE
modules/tectonic: Add cluster-config.yaml

### PR DESCRIPTION
**Pull request by yifan-gu**

### CODE REVIEW

**Comment by yifan-gu** [comment]
These are the flags we are using today in ingress, identity, stats-emitter, console. But I am not sure everything here is allowed to be changed by the users, so maybe we will remove some of those here.
cc @aaronlevy @diegs @derekparker @rithujohn191 @amrutac @squat

File: modules/tectonic/resources/manifests/utility-config.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: utility-config
```
Comment (diegs) [inline modules/tectonic/resources/manifests/utility-config.yaml]: I thought we were putting all of these in one `ConfigMap`, which is why you renamed it in #1928?

Comment (yifan-gu) [reply modules/tectonic/resources/manifests/utility-config.yaml]: @diegs That's good question. The reason is that then I realized that the bootkube module doesn't have all these required terraform variables defined.
So I couldn't decide if we want to put all those tectonic related terraform variables into bookube module.

Comment (diegs) [reply modules/tectonic/resources/manifests/utility-config.yaml]: Hm, maybe we should make this a `cluster-config` in the `tectonic-system` namespace and make the bootkube module `cluster-config` in the `kube-system` namespace then?

That way we can try and limit our number of config objects to one per module.

Comment (yifan-gu) [reply modules/tectonic/resources/manifests/utility-config.yaml]: @diegs I am ok with that, wdty @aaronlevy ?

File: modules/tectonic/resources/manifests/utility-config.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      grafana_callback: ${grafana_callback}
+      kubectl_client_id: ${kubectl_client_id}
+      kubectl_secret: ${kubectl_secret}
+      oidc_issuer_url: ${oidc_issuer_url}
+      prometheus_callback: ${prometheus_callback}
+    ingressConfig: |
+      console_base_host: ${console_base_host}
+    statsEmitterConfig: |
+      stats_url: ${stats_url}
+    tectonicConfigMap: |
```
Comment (yifan-gu) [inline modules/tectonic/resources/manifests/utility-config.yaml]: I feel we can actually dissolve key-values under this key into the above sub-keys.
I can do an investigation on who is using these values. But if someone knows this already, that will be helpful.
/cc @amrutac @squat @rithujohn191

File: modules/tectonic/resources/manifests/utility-config.yaml (hunk 1 above)
Comment (aaronlevy) [reply modules/tectonic/resources/manifests/utility-config.yaml]: Hmmm interesting thought. I agree that components should not need to cross namespace boundaries for their config (ideally). I'd be fine starting with this - and during the next few weeks of burn-in we can decide if we like that pattern.

File: modules/tectonic/resources/manifests/utility-config.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+metadata:
+  name: utility-config
+  namespace: tectonic-system
+data:
+  tuo-config: |
+    apiVersion: v1
+    kind: UtilityOperatorConfig
+    identityConfig: |
+      admin_email: ${admin_email}
+      admin_password_hash: ${admin_password_hash}
```
Comment (aaronlevy) [inline modules/tectonic/resources/manifests/utility-config.yaml]: Do we need this? I'd prefer that we don't keep anything sensitive in this config (also why do we need this - what component consumes?)

Comment (yifan-gu) [reply modules/tectonic/resources/manifests/utility-config.yaml]: The identity configmap already contains it: https://github.com/coreos/tectonic-installer/blob/master/modules/tectonic/resources/manifests/identity/configmap.yaml

File: modules/tectonic/resources/manifests/cluster-config.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      alertmanager_callback: ${alertmanager_callback}
+      console_callback: ${console_callback}
+      console_client_id: ${console_client_id}
+      console_secret: ${console_secret}
+      grafana_callback: ${grafana_callback}
+      kubectl_client_id: ${kubectl_client_id}
+      kubectl_secret: ${kubectl_secret}
+      oidc_issuer_url: ${oidc_issuer_url}
+      prometheus_callback: ${prometheus_callback}
+    ingressConfig: |
```
Comment (aaronlevy) [inline modules/tectonic/resources/manifests/cluster-config.yaml]: I'm on fence if this should be kube-addon or tectonic-utility. It's in the tectonic-system namespace - so I guess it makes sense here.

File: modules/bootkube/resources/manifests/cluster-config.yaml [hunk 5]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   name: cluster-config
-  namespace: tectonic-system
+  namespace: kube-system
```
Comment (aaronlevy) [inline modules/bootkube/resources/manifests/cluster-config.yaml]: wouldn't this be tectonic-system if it's managing components in that namespace?

Comment (diegs) [reply modules/bootkube/resources/manifests/cluster-config.yaml]: this one is in the `bootkube` module

Comment (aaronlevy) [reply modules/bootkube/resources/manifests/cluster-config.yaml]: nm. mixing up files

**Comment by diegs** [comment]
Can you remove `modules/bootkube/resources/manifests/01-tectonic-namespace.yaml` ?

File: modules/tectonic/resources/manifests/utility-config.yaml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      admin_user_id: ${admin_user_id}
+      alertmanager_callback: ${alertmanager_callback}
+      console_callback: ${console_callback}
+      console_client_id: ${console_client_id}
+      console_secret: ${console_secret}
+      grafana_callback: ${grafana_callback}
+      kubectl_client_id: ${kubectl_client_id}
+      kubectl_secret: ${kubectl_secret}
+      oidc_issuer_url: ${oidc_issuer_url}
+      prometheus_callback: ${prometheus_callback}
```
Comment (rithujohn191) [inline modules/tectonic/resources/manifests/utility-config.yaml]: If we plan to remove all the `callbacks` from the [IdentityConfig](https://github.com/coreos-inc/tectonic-utility-operator/pull/8/files) we should remove it from here as well

Comment (yifan-gu) [reply modules/tectonic/resources/manifests/utility-config.yaml]: will do.

File: modules/bootkube/resources/manifests/cluster-config.yaml (hunk 5 above)
Comment (rithujohn191) [reply modules/bootkube/resources/manifests/cluster-config.yaml]: So if we plan to make this ConfigMap general to all, how can I include my tectonic network config that's meant to be in the tectonic-system namespace?

File: modules/tectonic/resources/tectonic.sh [hunk 7]
```diff
@@ -127,6 +127,9 @@ kubectl create -f rbac/role-user.yaml
 kubectl create -f rbac/binding-admin.yaml
 kubectl create -f rbac/binding-discovery.yaml
 
+echo "Creating Tectonic Utility Operator Config"
+kubectl create -f utility-config.yaml
```
Comment (diegs) [inline modules/tectonic/resources/tectonic.sh]: `cluster-config.yaml` and update `echo` text?

**Comment by yifan-gu** [comment]
@rithujohn191 @diegs Addressed comments.

File: modules/tectonic/assets.tf [hunk 8]
```diff
@@ -58,6 +58,7 @@ resource "template_dir" "tectonic" {
     admin_email         = "${lower(var.admin_email)}"
     admin_password_hash = "${var.admin_password_hash}"
 
+    base_address         = "${var.base_address}"
     console_base_address = "${var.base_address}"
     console_base_host    = "${element(split(":", var.base_address), 0)}"
```
Comment (yifan-gu) [inline modules/tectonic/assets.tf]: @alexsomesan @s-urbaniak Does anyone know why this is not just `"${var.base_address}"`?

**Comment by yifan-gu** [comment]
Let me close and reopen the PR again to see if the smoke tests get running.

**Comment by yifan-gu** [comment]
Merging for now to unblock other works, we can address https://github.com/coreos/tectonic-installer/pull/1933#discussion_r141467260 later.

**Pull request merged**